plugins/ci/skills/payload-snapshot/scripts/payload_snapshot.py

```
@@ -122,11 +124,40 @@ class JobInfo:
-def fetch_json(url: str, timeout: int = 30) -> dict:
-    """Fetch JSON from a URL. Raises on error."""
+def _retry_delay(attempt: int, http_error=None) -> float:
+    """Compute retry delay: use Retry-After for 429s, otherwise exponential backoff."""
+    if http_error and http_error.code == 429:
+        retry_after = http_error.headers.get("Retry-After")
+        if retry_after:
+            try:
+                return float(retry_after)
+            except ValueError:
+                pass
+    return 2 ** (attempt + 1) + random.uniform(0, 1)
```

Ruff (0.15.15)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)